lib/http-client.ts

```diff
@@ -6,14 +6,41 @@ import {ResponsePromise} from "./promisewrappers";
 import Deferred = protractor.promise.Deferred;
 import Promise = protractor.promise.Promise;
 
-const controlFlow: protractor.promise.ControlFlow = protractor.promise.controlFlow();
+const controlFlow:protractor.promise.ControlFlow = protractor.promise.controlFlow();
+
+export interface IendPointTemplate {
```

Comment:
small nitpicking: the camelCase here is a bit weird, and the `I` prefix is not used anywhere else, it could be just `EndpointTemplate` IMO